extensions/namespace-lookup/src/test/java/io/druid/server/namespace/cache/JDBCExtractionNamespaceTest.java

```diff
@@ -285,24 +385,40 @@ public void testFindNew()
         keyName,
         valName,
         tsColumn,
-        new Period(1)
+        new Period(10)
     );
     extractionCacheManager.schedule(extractionNamespace);
     while (!fnCache.containsKey(extractionNamespace.getNamespace())) {
       Thread.sleep(1);
+      Assert.assertTrue("Timeout waiting on fn cache update", System.currentTimeMillis() - startTime < 30_000);
     }
     Function<String, String> extractionFn = fnCache.get(extractionNamespace.getNamespace());
     Assert.assertEquals(
+        "sanity check",
         "bar",
         extractionFn.apply("foo")
     );
 
     insertValues("foo", "baz", "2900-01-01 00:00:00");
-    Thread.sleep(100);
+
+    startTime = System.currentTimeMillis();
+    long pre = 0L;
```

Comment:
this code looks similar to the one above, can we create a function for it and explain what it does?